t/03-proxy_wasm/hfuncs/metrics/002-get_metric_misuse.t

```diff
@@ -0,0 +1,32 @@
+# vim:set ft= ts=4 sts=4 sw=4 et fdm=marker:
+
+use strict;
+use lib '.';
+use t::TestWasmX;
+
+plan_tests(7);
+run_tests();
+
+__DATA__
+
+=== TEST 1: proxy_wasm metrics - get_metric() invalid metric id
+--- valgrind
+--- load_nginx_modules: ngx_http_echo_module
+--- wasm_modules: hostcalls
+--- config
+    location /t {
+        proxy_wasm hostcalls 'on=request_headers \
+                              test=/t/metrics/get_invalid_metric';
+        echo ok;
+    }
+--- error_code: 500
+--- error_log eval
+[
+    qr/.+on_request_headers.+/,
```

Comment:
Another note about almost all of the new tests: it's not necessary to test for the step that will run the test, we can assert in this suite that the behavior of hostcalls is correct and the test will be executed in the intended step. We derive that from earlier tests in the suite that asserted the behavior was correct.

Reply:
d3c8dc22e88f5e38296390487525ce09b54ae286